Index: src/com/google/gwt/user/client/impl/DOMImplStandard.java
===================================================================
--- src/com/google/gwt/user/client/impl/DOMImplStandard.java	(revision 669)
+++ src/com/google/gwt/user/client/impl/DOMImplStandard.java	(revision 670)
@@ -172,14 +172,8 @@
 
     // Set up normal event dispatcher.
     $wnd.__dispatchEvent = function(evt) {
-      var listener, curElem = this;
-      while (curElem && !(listener = curElem.__listener))
-        curElem = curElem.parentNode;
-      if (curElem && curElem.nodeType != 1)
-        curElem = null;
-
-      if (listener)
-        @com.google.gwt.user.client.DOM::dispatchEvent(Lcom/google/gwt/user/client/Event;Lcom/google/gwt/user/client/Element;Lcom/google/gwt/user/client/EventListener;)(evt, curElem, listener);
+      if (this.__listener)
+        @com.google.gwt.user.client.DOM::dispatchEvent(Lcom/google/gwt/user/client/Event;Lcom/google/gwt/user/client/Element;Lcom/google/gwt/user/client/EventListener;)(evt, this, this.__listener);
     };
 
     $wnd.__captureElem = null;
