Index: src/com/google/gwt/user/client/impl/DOMImplMozilla.java
===================================================================
--- src/com/google/gwt/user/client/impl/DOMImplMozilla.java	(revision 16)
+++ src/com/google/gwt/user/client/impl/DOMImplMozilla.java	(working copy)
@@ -98,7 +98,13 @@
       if (parent.isSameNode(child)) {
         return true;
       }
-      child = child.parentNode;
+      try {
+        child = child.parentNode;
+      } catch(e) {
+        // Give up on 'Permission denied to get property HTMLDivElement.parentNode'
+        // See https://bugzilla.mozilla.org/show_bug.cgi?id=208427
+        return false;
+      }
       if (child.nodeType != 1) {
         child = null;
       }
