Index: src/com/google/gwt/user/client/impl/DOMImplStandard.java
===================================================================
--- src/com/google/gwt/user/client/impl/DOMImplStandard.java	(revision 726)
+++ src/com/google/gwt/user/client/impl/DOMImplStandard.java	(working copy)
@@ -200,7 +200,7 @@
       if (parent == child)
         return true;
       child = child.parentNode;
-      if (child.nodeType != 1)
+      if (child != null && child.nodeType != 1)
         child = null;
     }
     return false;
