Index: src/com/google/gwt/user/client/impl/DOMImplIE6.java
===================================================================
--- src/com/google/gwt/user/client/impl/DOMImplIE6.java	(revision 726)
+++ src/com/google/gwt/user/client/impl/DOMImplIE6.java	(working copy)
@@ -36,6 +36,18 @@
     return $doc.createElement("<INPUT type='RADIO' name='" + group + "'>");
   }-*/;
 
+  public native int eventGetClientX(Event evt) /*-{
+    // Standard mode use $doc.documentElement.clientLeft (= always 2)
+    // Quirks mode use $doc.body.clientLeft (=BODY border width)
+    return evt.clientX - ($doc.documentElement.clientLeft || $doc.body.clientLeft);
+  }-*/;
+
+  public native int eventGetClientY(Event evt) /*-{
+    // Standard mode use $doc.documentElement.clientTop (= always 2)
+    // Quirks mode use $doc.body.clientTop (= BODY border width)
+    return evt.clientY - ($doc.documentElement.clientTop || $doc.body.clientTop);
+  }-*/;
+
   public native Element eventGetTarget(Event evt) /*-{
     var elem = evt.srcElement;
     return elem ? elem : null;
