Index: src/com/google/gwt/user/client/ui/FlowPanel.java
===================================================================
--- src/com/google/gwt/user/client/ui/FlowPanel.java	(revision 307)
+++ src/com/google/gwt/user/client/ui/FlowPanel.java	(working copy)
@@ -55,6 +55,12 @@
     return getChildren().indexOf(child);
   }
 
+  public void insert(Widget w, int beforeIndex) {
+    w.removeFromParent();
+    super.insert(w, null, beforeIndex);
+    DOM.insertChild(getElement(), w.getElement(), beforeIndex);
+  }
+
   public boolean remove(int index) {
     return remove(getWidget(index));
   }
