Index: user/src/com/google/gwt/user/client/ui/ScrollPanel.java
===================================================================
--- user/src/com/google/gwt/user/client/ui/ScrollPanel.java	(revision 861)
+++ user/src/com/google/gwt/user/client/ui/ScrollPanel.java	(working copy)
@@ -32,6 +32,8 @@
   public ScrollPanel() {
     setAlwaysShowScrollBars(false);
     sinkEvents(Event.ONSCROLL);
+    // Prevent IE standard mode bug when AbsolutePanel is among our children
+    DOM.setStyleAttribute(getElement(), "position", "relative");
   }
 
   /**
