Index: src/com/google/gwt/user/client/impl/DOMImplMozilla.java
===================================================================
--- src/com/google/gwt/user/client/impl/DOMImplMozilla.java	(revision 16)
+++ src/com/google/gwt/user/client/impl/DOMImplMozilla.java	(working copy)
@@ -60,6 +60,11 @@
       parent = parent.parentNode;
     }
  
+  var borderLeftWidth = $doc.defaultView.getComputedStyle(elem, null).getPropertyValue("border-left-width");
+  if (borderLeftWidth.indexOf("px") != -1) {
+    left = left - parseInt(borderLeftWidth.substr(0, borderLeftWidth.length - 2));
+  }
+
   // Must cover both Standard and Quirks mode. 
     return left + $doc.body.scrollLeft + $doc.documentElement.scrollLeft;
   }-*/;
@@ -75,6 +80,11 @@
       parent = parent.parentNode;
     }
    
+  var borderTopWidth = $doc.defaultView.getComputedStyle(elem, null).getPropertyValue("border-top-width");
+  if (borderTopWidth.indexOf("px") != -1) {
+    top = top - parseInt(borderTopWidth.substr(0, borderTopWidth.length - 2));
+  }
+
     // Must cover both Standard and Quirks mode.
     return top + $doc.body.scrollTop + $doc.documentElement.scrollTop;
   }-*/;
