Index: src/com/google/gwt/user/client/impl/DOMImplIE6.java
===================================================================
--- src/com/google/gwt/user/client/impl/DOMImplIE6.java	(revision 726)
+++ src/com/google/gwt/user/client/impl/DOMImplIE6.java	(working copy)
@@ -62,9 +62,8 @@
       scrollLeft = $doc.body.scrollLeft
     }
     
-    // Offset needed as IE starts the window's upper left at
-    // 2,2 rather than 0,0.
-    return (elem.getBoundingClientRect().left + scrollLeft) - 2;
+    return (elem.getBoundingClientRect().left + scrollLeft)
+		    - ($doc.documentElement.clientLeft || $doc.body.clientLeft);
   }-*/;
 
   public native int getAbsoluteTop(Element elem) /*-{
@@ -75,9 +74,8 @@
       scrollTop = $doc.body.scrollTop
     } 
     
-    // Offset needed as IE starts the window's upper left as 2,2 
-    // rather than 0,0.
-    return (elem.getBoundingClientRect().top +  scrollTop) - 2;
+    return (elem.getBoundingClientRect().top +  scrollTop)
+		    - ($doc.documentElement.clientTop || $doc.body.clientTop);
    }-*/;
 
   public native Element getChild(Element elem, int index) /*-{
